javacutil/src/main/java/org/checkerframework/javacutil/CollectionUtils.java

```diff
@@ -23,4 +28,203 @@ protected boolean removeEldestEntry(Map.Entry<K, V> entry) {
       }
     };
   }
+
+  ///
+  /// Temporary utility methods
+  ///
+
+  // TODO: After code review, copy these methods into plume-util.
+  // After plume-util 1.6.6 is released, use these methods from it.
+
+  /**
+   * Returns a copy of {@code orig}, where each element of the result is a clone of the
+   * corresponding element of {@code orig}.
+   *
+   * @param <T> the type of elements of the collection
+   * @param <C> the type of the collection
+   * @param orig a collection
+   * @return a deep copy of {@code orig}
+   */
+  @SuppressWarnings({
+    "signedness", // problem with clone()
+    "nullness:return" // "return null;"
+  })
+  public static <@Nullable T, C extends @Nullable Collection<T>> C cloneElements(C orig) {
+    if (orig == null) {
+      return null;
+    }
+    C result = clone(orig);
+    result.clear();
+    for (T elt : orig) {
+      result.add(clone(elt));
+    }
+    return result;
+  }
+
+  /**
+   * Returns a copy of {@code orig}, where each key and value in the result is a clone of the
+   * corresponding element of {@code orig}.
+   *
+   * @param <K> the type of keys of the map
+   * @param <V> the type of values of the map
+   * @param <M> the type of the map
+   * @param orig a map
+   * @return a deep copy of {@code orig}
+   */
+  @SuppressWarnings({"nullness", "signedness"}) // generics problem with clone
+  public static <K, V, M extends Map<K, V>> M cloneElements(M orig) {
+    return cloneElements(orig, false);
```

Comment:
I think this should be `cloneElements(orig, true)`, according to the JavaDoc. As it is, the implementation is also identical to `cloneValues`, which seems wrong.